bundle/edu.gemini.spdb.reports.collection/src/main/java/edu/gemini/dbTools/odbState/ObservationState.java

```diff
@@ -144,42 +143,38 @@ public TooType getTooType() {
     }
 
     /**
-     * Gets a formated string that indicates the night during which the
+     * Gets a formatted string that indicates the night during which the
      * observation took place.
      * @param time utc time of the last observation event message
      * @return formatted string indicating the night during which the
      * observation took place.
      */
     private static String _getNight(final long time) {
-        final Calendar cal = GregorianCalendar.getInstance();
-        cal.setTimeInMillis(time);
-
-        // The calendar is in the local time zone.  If the time is before
-        // noon, then the night started on the previous day.  If the time
-        // is after noon, then the night ends on the next day.
-        final int ampm = cal.get(Calendar.AM_PM);
-
-        final Calendar endCal;
-
-        final int startDay;
-        final int endDay;
-        if (ampm == Calendar.AM) {
-            endCal = (Calendar) cal.clone();
-            endDay = cal.get(Calendar.DAY_OF_MONTH);
-            cal.add(Calendar.DAY_OF_MONTH, -1);
-            startDay = cal.get(Calendar.DAY_OF_MONTH);
+        // Determine when, in the local time zone, the time started:
+        // If the time is before noon, then the night started on the previous day.
+        // If the time is after noon, then the night ends on the next day.
+        // Note: AM is 0, PM is 1 by documentation for ChronoField.
+        final LocalDateTime dt = LocalDateTime.ofInstant(Instant.ofEpochMilli(time), ZoneId.systemDefault());
+        final int ampm = dt.get(ChronoField.AMPM_OF_DAY);
+
+        final LocalDateTime start;
+        final LocalDateTime end;
+        if (ampm == 0) {
+            start = dt.minus(1, ChronoUnit.DAYS);
+            end   = dt;
         } else {
-            startDay = cal.get(Calendar.DAY_OF_MONTH);
-            cal.add(Calendar.DAY_OF_MONTH, 1);
-            endCal = cal;
-            endDay = cal.get(Calendar.DAY_OF_MONTH);
+            start = dt;
+            end   = dt.plus(1, ChronoUnit.DAYS);
         }
 
+        final int startDay = start.get(ChronoField.DAY_OF_MONTH);
+        final int endDay   = end.get(ChronoField.DAY_OF_MONTH);
+
         final StringBuilder buf = new StringBuilder();
         buf.append(startDay).append("/").append(endDay).append(" ");
 
-        final DateFormat f = new SimpleDateFormat("MMM yyyy");
-        buf.append(f.format(endCal.getTime()));
+        final DateTimeFormatter f = DateTimeFormatter.ofPattern("MMM yyyy").withZone(ZoneId.of("UTC"));
+        buf.append(f.format(end));
 
         return buf.toString();
     }
```

Comment:
I was slightly worried that I might have screwed something up here, so I executed the following test to compare the old and new output strings from now to 10,000 hours from now, which is over 416 days, and the comparisons always succeeded.

```scala
import java.time._
import java.time.temporal._
import java.time.format._
import java.text._
import java.util._

def tOld(time: Long): String = {
  val dt = LocalDateTime.ofInstant(Instant.ofEpochMilli(time), ZoneId.systemDefault())
  val ampm = dt.get(ChronoField.AMPM_OF_DAY)

  val start = if (ampm == 0) dt.minus(1, ChronoUnit.DAYS) else dt
  val end   = if (ampm == 0) dt else dt.plus(1, ChronoUnit.DAYS)

  val startDay = start.get(ChronoField.DAY_OF_MONTH)
  val endDay   = end.get(ChronoField.DAY_OF_MONTH)

  val df = DateTimeFormatter.ofPattern("MMM yyyy").withZone(ZoneId.systemDefault())
  s"$startDay/$endDay ${df.format(end)}"
}

def tNew(time: Long): String = {
  val cal = Calendar.getInstance()
  cal.setTimeInMillis(time.toLong)

  // The calendar is in the local time zone.  If the time is before
  // noon, then the night started on the previous day.  If the time
  // is after noon, then the night ends on the next day.
  val ampm: Int = cal.get(Calendar.AM_PM)

  var endCal: Calendar = null

  var startDay: Int = 0
  var endDay: Int = 0
  if (ampm == Calendar.AM) {
    endCal = cal.clone.asInstanceOf[Calendar]
    endDay = cal.get(Calendar.DAY_OF_MONTH)
    cal.add(Calendar.DAY_OF_MONTH, -1)
    startDay = cal.get(Calendar.DAY_OF_MONTH)
  }
  else {
    startDay = cal.get(Calendar.DAY_OF_MONTH)
    cal.add(Calendar.DAY_OF_MONTH, 1)
    endCal = cal
    endDay = cal.get(Calendar.DAY_OF_MONTH)
  }

  val f: DateFormat = new SimpleDateFormat("MMM yyyy")
  s"$startDay/$endDay ${f.format(endCal.getTime)}"
}

for (i <- Range(0,10000)) {
  val time = Instant.now().plus(i, ChronoUnit.HOURS).toEpochMilli
  assert(tOld(time).equals(tNew(time)))
}
```

Reply:
The `var endCal` is because I copy / pasted from Java and allowed it to just convert. ☺️